### PR TITLE
Tidying and msg.c work

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: include/common_structs.h [hunk 1]
```diff
@@ -1057,39 +1057,98 @@ typedef struct SpriteComponent {
...
+    /* 0x524 */ s32 speedSoundIDB;
+    /* 0x528 */ s16 varBufferReadPos;
+    /* 0x52A */ s8 unk_52A;
     /* 0x52B */ u8 currentImageIndex;
-    /* 0x52C */ char unk_52C[10];
-    /* 0x536 */ s16 height;
-    /* 0x538 */ s16 width;
-    /* 0x53A */ char unk_53A[30];
-} PrintContext; // size = 0x558
+    /* 0x52C */ Vec2s varImgeScreenPos; // in addition, posX=0 is taken as 'dont draw'
```
Comment (nanaian) [inline include/common_structs.h:1133]: "Imge"?

File: src/msg.c [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+INCLUDE_ASM(s32, "msg", _update_message, MessagePrintState* msgPrintState);
+#endif
+
+INCLUDE_ASM(s32, "msg", render_messages);
+
+void msg_play_speech_sound(MessagePrintState* printer, u8 character) {
+    f32 volTemp;
+    s16 volume;
+    s32 pitchShift;
+    s32 flag = 1;
```
Comment (nanaian) [inline src/msg.c:458]: Weird that this is required

File: src/msg.c [hunk 3]
```diff
@@ -0,0 +1,721 @@
...
+    }
+}
+
+INCLUDE_ASM(s32, "msg", msg_copy_to_print_buffer);
+
+void initialize_printer(MessagePrintState* printer, s32 arg1, s32 arg2) {
+    s32 i;
+
+    printer->printBufferSize = ARRAY_COUNT(printer->printBuffer);
+    printer->printBuffer[0] = 0xFB;
```
Comment (nanaian) [inline src/msg.c:484]: Worth noting - possibly through a comment? - that this is `[next]`; we should probably make an enum for all the characters/opcodes (see tools/build/msg/parse_compile.py).

**Review by nanaian — APPROVED** [review]

File: src/msg.c (hunk 2 above)
Comment (ethteck) [reply src/msg.c:458]: indeed

**Review by nanaian — APPROVED** [review]

**Pull request merged**